tests/unit/test-cmd/test.build.js

```diff
@@ -64,313 +56,269 @@ describe('build', () => {
-          .then((buildResult) => zipFile.open(buildResult.extensionPath))
-          .then(() => zipFile.extractFilenames())
-          .then((fileNames) => {
-            assert.notInclude(fileNames, 'background-script.js');
-            return zipFile.close();
-          })
-    );
-  });
+  it('asks FileFilter what files to include in the ZIP', () => withTempDir(
+    async (tmpDir) => {
+      const zipFile = new ZipFile();
+      const fileFilter = new FileFilter({
+        sourceDir: '.',
+        baseIgnoredPatterns: ['**/background-script.js'],
+      });
+      const buildResult = await build({
+        sourceDir: fixturePath('minimal-web-ext'),
+        artifactsDir: tmpDir.path(),
+      }, {fileFilter});
+      await zipFile.open(buildResult.extensionPath);
+      const fileNames = await zipFile.extractFilenames();
+      assert.notInclude(fileNames, 'background-script.js');
+      await zipFile.close();
+    }
+  ));
 
   it('lets you rebuild when files change', () => withTempDir(
-    (tmpDir) => {
+    async (tmpDir) => {
       const sourceDir = fixturePath('minimal-web-ext');
       const artifactsDir = tmpDir.path();
       const fileFilter = new FileFilter({sourceDir, artifactsDir});
       sinon.spy(fileFilter, 'wantFile');
       const onSourceChange = sinon.spy(() => {});
-      return build({
+      const buildResult = await build({
         sourceDir, artifactsDir, asNeeded: true,
       }, {
         manifestData: basicManifest, onSourceChange, fileFilter,
-      })
-        .then((buildResult) => {
-          // Make sure we still have a build result.
-          assert.match(buildResult.extensionPath, /\.zip$/);
-          return buildResult;
-        })
-        .then((buildResult) => {
-          const args = onSourceChange.firstCall.args[0];
-
-          sinon.assert.called(onSourceChange);
-          sinon.assert.calledWithMatch(onSourceChange, {
-            artifactsDir,
-            sourceDir,
-          });
-
-          assert.typeOf(args.onChange, 'function');
-
-          // Make sure it uses the file filter.
-          assert.typeOf(args.shouldWatchFile, 'function');
-          args.shouldWatchFile('/some/path');
-          sinon.assert.called(fileFilter.wantFile);
-
-          // Remove the built extension.
-          return fs.unlink(buildResult.extensionPath)
-            // Execute the onChange handler to make sure it gets built
-            // again. This simulates what happens when the file watcher
-            // executes the callback.
-            .then(() => args.onChange());
-        })
-        .then((buildResult) => {
-          assert.match(buildResult.extensionPath, /\.zip$/);
-          return fs.stat(buildResult.extensionPath);
-        })
-        .then((stat) => {
-          assert.equal(stat.isFile(), true);
-        });
+      });
+      // Make sure we still have a build result.
+      assert.match(buildResult.extensionPath, /\.zip$/);
+      assert.equal(onSourceChange.called, true);
```

Comment:
we rewrote some of these kind of assertion to use the sinon assertions some times ago (http://sinonjs.org/releases/v4.1.4/assertions/, e.g. an assertion in the form `assert.equal(aSinonSpy.called, true)` has been rewritten into `sinon.assert.called(aSinonSpy);`).

Let's restore the usage of the sinon assertions here (also for the other assertions that were using the sinon assertions in the previous version)